terraform/node/k8s/deploy/main.tf

```diff
@@ -0,0 +1,172 @@
+
+      # Clone and install operator onto cluster
+      echo "LOG: Cloning helm-charts repo"
+      # TODO: Update to latest release version of the code
+      git clone https://github.com/aws-observability/helm-charts -q
+      cd helm-charts/charts/amazon-cloudwatch-observability/
+      git reset --hard b407a9545a03e377703ef965a2ee655aa6df7406
+
+      echo "LOG: Installing CloudWatch Agent Operator using Helm"
+      helm upgrade --install --debug --namespace amazon-cloudwatch amazon-cloudwatch-operator ./ --create-namespace --set region=${var.aws_region} --set clusterName=k8s-cluster-${var.test_id}
+
+      # Wait for pods to exist before checking if they're ready
+      sleep 60
+      kubectl wait --for=condition=Ready pods --all --selector=app.kubernetes.io/name=amazon-cloudwatch-observability -n amazon-cloudwatch --timeout=60s
+      kubectl wait --for=condition=Ready pods --all --selector=app.kubernetes.io/name=cloudwatch-agent -n amazon-cloudwatch --timeout=60s
+
+      if [ "${var.repository}" = "amazon-cloudwatch-agent" ]; then
+        RELEASE_TESTING_SECRET_NAME=release-testing-ecr-secret
+        RELEASE_TESTING_TOKEN=`aws ecr --region=us-west-2 get-authorization-token --output text --query authorizationData[].authorizationToken | base64 -d | cut -d: -f2`
+        kubectl delete secret -n amazon-cloudwatch --ignore-not-found $RELEASE_TESTING_SECRET_NAME
+        kubectl create secret -n amazon-cloudwatch docker-registry $RELEASE_TESTING_SECRET_NAME \
+          --docker-server=https://${var.release_testing_ecr_account}.dkr.ecr.us-west-2.amazonaws.com \
+          --docker-username=AWS \
+          --docker-password="$${RELEASE_TESTING_TOKEN}"
+
+        kubectl patch serviceaccount cloudwatch-agent -n amazon-cloudwatch -p='{"imagePullSecrets": [{"name": "release-testing-ecr-secret"}]}'
+        kubectl delete pods --all -n amazon-cloudwatch
+      elif [ "${var.repository}" = "amazon-cloudwatch-agent-operator" ]; then
+        RELEASE_TESTING_SECRET_NAME=release-testing-ecr-secret
+        RELEASE_TESTING_TOKEN=`aws ecr --region=us-west-2 get-authorization-token --output text --query authorizationData[].authorizationToken | base64 -d | cut -d: -f2`
+        kubectl delete secret -n amazon-cloudwatch --ignore-not-found $RELEASE_TESTING_SECRET_NAME
+        kubectl create secret -n amazon-cloudwatch docker-registry $RELEASE_TESTING_SECRET_NAME \
+          --docker-server=https://${var.release_testing_ecr_account}.dkr.ecr.us-west-2.amazonaws.com \
+          --docker-username=AWS \
+          --docker-password="$${RELEASE_TESTING_TOKEN}"
+
+        kubectl patch deploy -n amazon-cloudwatch amazon-cloudwatch-observability-controller-manager --type='json' -p='[{"op": "add", "path": "/spec/template/spec/imagePullSecrets", "value": [{"name": "release-testing-ecr-secret"}]}]'
+        kubectl delete pods --all -n amazon-cloudwatch
+      fi
+
+      if [ "${var.repository}" = "amazon-cloudwatch-agent-operator" ]; then
```

Comment:
This if condition is the same as the one in line 64. This looks confusing. Could you add some comments to explain?

Reply:
Will add some comments, but this was copied from other workflows and as stated in another comment I will confirm functionality of release testing when finalizing the e2e tests